File: src/main/resources/output/IFC4X3_js/Entities/IfcCoordinateReferenceSystem/IfcProjectedCRS/IfcProjectedCRS.js
class IfcProjectedCRS extends IfcCoordinateReferenceSystem {
    constructor() {
        /** @type {IFCIDENTIFIER} */
        this.VerticalDatum = null;
        /** @type {IFCIDENTIFIER} */
        this.MapProjection = null;
        /** @type {IFCIDENTIFIER} */
        this.MapZone = null;
        /** @type {IFCNAMEDUNIT} */
        this.MapUnit = null;
    }

    // === WHERE CLAUSES ===
    // MapUnitIsLength : NOT(EXISTS(SELF.MapUnit)) OR (SELF.MapUnit.UnitType = IfcUnitEnum.LENGTHUNIT)
}

File: src/main/resources/output/IFC4X3_java/Entities/IfcCoordinateReferenceSystem/IfcProjectedCRS/IfcProjectedCRS.java
public class IfcProjectedCRS extends IfcCoordinateReferenceSystem {
    public IfcIdentifier VerticalDatum;
    public IfcIdentifier MapProjection;
    public IfcIdentifier MapZone;
    public IfcNamedUnit MapUnit;

    // === WHERE CLAUSES ===
    // MapUnitIsLength : NOT(EXISTS(SELF.MapUnit)) OR (SELF.MapUnit.UnitType = IfcUnitEnum.LENGTHUNIT)
}
